compiler/src/org.graalvm.compiler.phases.common/src/org/graalvm/compiler/phases/common/UseTrappingNullChecksPhase.java

```diff
@@ -287,4 +292,22 @@ private static void replaceWithTrappingNullCheck(AbstractDeoptimizeNode deopt, I
         GraphUtil.killCFG(trappingContinuation);
         GraphUtil.tryKillUnused(isNullNode);
     }
+
+    private static NullCheckNode tryUseUncompressedNullCheck(ValueNode value, IsNullNode isNullNode) {
+        assert isNullNode.getValue().equals(value);
+        CompressionNode uncompressed = null;
+        for (Node usage : value.usages()) {
+            if (usage instanceof CompressionNode) {
+                if (((CompressionNode) usage).getOp() == CompressionNode.CompressionOp.Uncompress) {
+                    assert ((CompressionNode) usage).getValue().equals(value);
+                    uncompressed = (CompressionNode) usage;
```

Comment:
Couldn't there be multiple uncompress usages?

Reply:
I searched the codes that the `uncompress` is mostly generated together with its `input`. And If there are multiple uncompress usages, would they be optimized to a single one by GVN? If it exists the corner cases, I think we'd better add the restriction of the single usage of uncompress.

Reply:
Yes, please add the restriction and only apply the transformation if there is a single uncompress usage.

Reply:
OK! Thank you for review! I will update soon!

Reply:
Hi @dougxc I'v updated the patch according to your comment. So could you please take a look at it again? Thanks!

Reply:
Looks good. I've started the process of integration.

Reply:
Thanks again for your review!